src/app/services/ui.ts

```diff
@@ -35,8 +35,9 @@ export class UIService {
   }
 
   setProfile(profile: NostrProfileDocument | undefined) {
+    const before = this.#profile
     this.#profile = profile;
-    this.#profileChanged.next(this.#profile);
+    if (JSON.stringify(before) !== JSON.stringify(profile)) this.#profileChanged.next(this.#profile);
```

Comment:
I initially thought one alternative could be to use the event ID (not available on profile right now), but it might be some local metadata changes to the profile. We'll go with this, I presume you've tested and experienced that this helps.

Reply:
thanks! 🤙🏻